### PR TITLE
fix(MintTokens): Network is not correctly initialized when the new token form is prefilled

### LINKED ISSUE #11030: [Community Mint Tokens] Network is not correctly initialized when the new token form is prefilled
### Description
When a token deployment fails, the user is able to `Retry`, that means the view changes from the `CommunityTokenView` to the `CommunityNewTokenView` form, and this new token view form is prefilled with the existing token information.

Network information is not correctly initialized since now it takes the first element in the combobox, instead of selecting the correct `chainId`.

Anaylize how `NetwordFileter.qml` initialization works and if it is kind of coupled with wallet funcionality, create a simple network combobox for this view. 

**NOTE**: Start development from: https://github.com/status-im/status-desktop/pull/10974

### PR DESCRIPTION
**Pull request by noeliaSD**
Fixes #11030 

### What does the PR do

- It updates `NetworkFilter.qml` to allow setting a specific `chainId` when single selection mode is on.
- It adds `storybook` use case for setting specific `chainId` in single selection mode.
- It integrates new network initialization process in mint new token view form to initialize network combobox properly.

### Affected areas

Community Settings / Mint Tokens

### Screenshot of functionality 

- App integration:

https://github.com/status-im/status-desktop/assets/97019400/a97052c0-ba51-4fd8-bdd4-9895e853e5cc

- Storybook usecase:

https://github.com/status-im/status-desktop/assets/97019400/421ebb6d-59dd-4280-b4e2-7d9ab9636c44

### CODE REVIEW

**Review by stefandunca — APPROVED** [review]
Good job!

File: ui/app/AppLayouts/Wallet/controls/NetworkFilter.qml [hunk 1]
```diff
@@ -37,16 +62,21 @@ StatusComboBox {
...
-    Component.onCompleted: {
-        if (!multiSelection && d.currentModel.count > 0) {
-            d.selectedChainName = d.currentModel.rowData(d.currentIndex, "chainName")
-            d.selectedIconUrl = d.currentModel.rowData(d.currentIndex, "iconUrl")
+        function setChainProperties(networkModel, index) {
+            // Update selected chain properties:
+            d.currentModel = networkModel
+            d.currentIndex = index
+            d.selectedChainName = ModelUtils.get(d.currentModel, d.currentIndex, "chainName")
+            d.selectedIconUrl = ModelUtils.get(d.currentModel, d.currentIndex, "iconUrl")
```
Comment (alexjba) [inline ui/app/AppLayouts/Wallet/controls/NetworkFilter.qml:71]: WDYT of making these properties readonly and use only bindings? Should keep us safe from inconsistency issues.

**Review by caybro — APPROVED** [review]
LGTM

File: ui/app/AppLayouts/Wallet/controls/NetworkFilter.qml (hunk 1 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Wallet/controls/NetworkFilter.qml:71]: Good idea!

**Pull request merged**